src/args/make-script/template.sh

```diff
@@ -24,6 +24,22 @@ function prompt_user_for_confirmation {
     fi
 }
 
+function prompt_user_for_input {
+  local answer
+  local default="${1:-}"
+
+  warn Please enter your input: \
+    && if running_in_ci_cd_provider; then
+      critical Running on CI/CD provider, cannot ask for user input, there are no humans around
+    else
+      read -ep '>>> ' -r answer \
+        && if test -z "${answer:default}"; then
```

Comment:
this is not doing the fallback to $default

![image](https://user-images.githubusercontent.com/47480384/134235910-f1eab518-440a-402d-8012-b7abe8cbcc9f.png)

if you want that on CI/CD the script stops immediately, we can avoid locally having a default value and directly echo back the user input

```bash
function prompt_user_for_input {
  local answer

  warn Please enter your input: \
    && if running_in_ci_cd_provider; then
      critical Running on CI/CD provider, cannot ask for user input, there are no humans around
    else
      read -ep '>>> ' -r answer \
        && echo "${answer}"
    fi
}
```

or the correct syntax for what you try to do in this PR:

```bash
function prompt_user_for_input {
  local answer
  local default="${1:-}"

  warn Please enter your input: \
    && if running_in_ci_cd_provider; then
      critical Running on CI/CD provider, cannot ask for user input, there are no humans around
    else
      read -ep '>>> ' -r answer \
        && if test -z "${answer}"; then
          warn empty input, assuming default value of: "${default}" \
           && echo "${default}"
        else
           echo "${answer}"
        fi
    fi
}
```

Reply:
I almost got something working, but it looks the critical exit is not propagated:

Reply:
See changes: https://github.com/fluidattacks/makes/compare/cb81996c2ee006abc169f16b514bb918cc4f9555..b15c107d73ae12931032e9445eff3d6148e7ca77